### ISSUE TITLE
Disallow view /users/<username> except for administrators

### ISSUE THREAD

**Issue by RobinIsTheBird** [OpenTreeMap/otm-core#2781]
Related to #136 

A user has asked to be able to close their account.

If the user has created or updated any features, udfs, etc, we really can't delete them.

We could add a new permission, "users who have contributed to this map", to be set in our roles. We could then give users the option of removing view privileges for their /users/<username> for anyone who doesn't have a role with that new permission set to "view" for any map the user contributed to.

**Comment by RickMohr** [comment]
Does this mean non-administrators can never look at another person's profile? Seems like a bad idea to me.

**Comment by RobinIsTheBird** [comment]
@RickMohr yes, but only if the user's account has that setting, either by the user setting it or by the administrator setting it. It's the closest I think we can come to deleting a user.
